src/Parser/Core/Modules/Items/BFA/Crafted/DarkmoonDeckBlockades.js

```diff
@@ -24,26 +51,125 @@ const DARKMOON_DECK_BLOCKADES_CARDS = [
 
 class DarkmoonDeckBlockades extends Analyzer {
   healing = 0;
+  actualStaminaIncreasePerCard = {};
+  cardTracker = {};
 
   constructor(...args) {
     super(...args);
     this.active = this.selectedCombatant.hasTrinket(ITEMS.DARKMOON_DECK_BLOCKADES.id);
+    if (this.active) {
+      Object.keys(DARKMOON_DECK_BLOCKADES_CARDS).forEach((buffId) => {
+        const defaultCard = DARKMOON_DECK_BLOCKADES_CARDS[buffId];
+        const actualStaminaIncrease = calculatePrimaryStat(355, defaultCard.staminaIncrease, this.selectedCombatant.getItem(ITEMS.DARKMOON_DECK_BLOCKADES.id).itemLevel);
+        this.actualStaminaIncreasePerCard[buffId] = actualStaminaIncrease;
+      });
+    }
+  }
+
+  _isBlockadesCard(spellId) {
+    const cardId = '' + spellId;
+    const darkmoonSpells = Object.keys(DARKMOON_DECK_BLOCKADES_CARDS);
+    if (darkmoonSpells.includes(cardId)) {
+      return true;
+    }
+
+    return false;
   }
 
   on_byPlayer_heal(event) {
-    const spellId = event.ability.guid;
-    if (!DARKMOON_DECK_BLOCKADES_CARDS.includes(spellId)) {
+    if (!this._isBlockadesCard(event.ability.guid)) {
       return;
     }
     this.healing += (event.amount || 0) + (event.absorbed || 0);
   }
 
+  on_toPlayer_applybuff(event) {
+    const spellId = event.ability.guid;
+    if (!this._isBlockadesCard(spellId)) {
+      return;
+    }
+    if (!this.cardTracker.hasOwnProperty(spellId)) {
+      this.cardTracker[spellId] = [];
+    }
+
+    this.cardTracker[spellId].push({
+      start: event.timestamp,
+      end: UNFINISHED,
+    });
+  }
+
+  on_toPlayer_removebuff(event) {
+    const spellId = event.ability.guid;
+    if (!this._isBlockadesCard(spellId)) {
+      return;
+    }
+    if (!this.cardTracker.hasOwnProperty(spellId)) {
+      return;
+    }
+    const cardsOfThisType = this.cardTracker[spellId];
+    cardsOfThisType.forEach((cardOccurance) => {
```

Comment:
This shouldn't be necessary, the last active card of this type is always the last pushed element (`cardsOfThisType[cardsOfThisType.length - 1]`).